dask/array/tests/test_creation.py

```diff
@@ -61,6 +61,15 @@ def test_arr_like(funcname, shape, cast_shape, dtype, cast_chunks, chunks):
         assert (np_r == np.asarray(da_r)).all()
 
 
+@pytest.mark.parametrize("x, y, z", [(100, 25, 25)])
+def test_arr_hashing(x, y, z):
+    # 3d numpy array
+    arr = np.random.rand(x, y, z)
+    with dask.config.set({"tokenize.warn_time_secs": 1e-5}):
```

Comment:
FWIW, I'm not sure that we ever want to warn for things we know how to tokenize efficiently... AFAIK, there's nothing the user can do to actually fix this warning, right? That differs from something like the "Large object in Dask graph" warning, which they can actually take action on.

Reply:
Though it's not clear to me how we can know whether or not we have an efficient tokenizer, at least not within `tokenize`. I'm not sure what's best here.

Reply:
We could have a special value like -1 (or another key) to indicate that we want to disable the check. For predicting the efficiency of the tokenizer, I think the depth of the recursion on `normalize_token` and which path it takes are important, as well as the size of the input. I am also planning to add an option to skip hashing numpy arrays and pandas dataframes per https://github.com/dask/dask/issues/4275.

Reply:
@TomAugspurger @mrocklin do you have any thoughts on this? We do have some idea about what can make the tokenizer less efficient, but we are not checking for it so far. I'd like to come to an understanding before moving forward.

Reply:
I'm not sure unfortunately. My gut reaction is that we shouldn't be warning users when there isn't a concrete action they can take to address that warning. I'm not sure where that leaves us w.r.t. that.

Reply:
@TomAugspurger that concern, I think, can be addressed with a new config key or a special value as I said above. Which do you prefer? Any thoughts on my response to your other concern (about the efficient tokenizers)?

Reply:
What would that config look like? Would the default be to warn, and the config provides an option for controlling that warning?

If we're able to disable this completely for things we know we can tokenize efficiently, even if still takes a while for large objects, then that's more palatable. Though in that case should you update the test to not use an ndarray?

Reply:
Yes, the default would be to warn the user. For example, tokenize.warn_time_secs: -1 disables the warning, and the warning message includes this information. 

We can check some function of total byte size, number of objects, and object type. The test could use the older version where the pathological ndarray needs pickling, so it is inefficient, while the regular ndarray takes half the time, but can still take a long time depending on size.

Reply:
I'll defer to others on this, but my general preference is to not issue a warning to users if there's nothing they can do to avoid it (other than configuring things to silence the warning).

> Though in that case should you update the test to not use an ndarray?

Will the test need to be updated? Will we ever warn for an ndarray?

Reply:
We will warn for an inefficient ndarray. For example, we can keep the test from the issue, and issue a warning when we need to pickle something, also when we detect inefficiency in other ways.